sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -683,6 +686,91 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
+        /// using this properties.
+        /// </remarks>
+        /// <exception cref="ServiceBusException">
+        ///   <list type="bullet">
+        ///     <item>
+        ///       <description>
+        ///         The lock for the message has expired or the message has already been completed. This does not apply for session-enabled entities.
+        ///         The <see cref="ServiceBusException.Reason" /> will be set to <see cref="ServiceBusFailureReason.MessageLockLost"/> in this case.
+        ///       </description>
+        ///     </item>
+        ///     <item>
+        ///       <description>
+        ///         The lock for the session has expired or the message has already been completed. This only applies for session-enabled entities.
+        ///         The <see cref="ServiceBusException.Reason" /> will be set to <see cref="ServiceBusFailureReason.SessionLockLost"/> in this case.
+        ///       </description>
+        ///     </item>
+        ///   </list>
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <list type="bullet">
+        ///     <item>
+        ///       <description>
+        ///         The <paramref name="message"/> argument was null.
+        ///       </description>
+        ///     </item>
+        ///     <item>
+        ///       <description>
+        ///         The dead letter reason or dead letter error exception was specified in both the parameter and the properties dictionary.
+        ///       </description>
+        ///     </item>
+        ///   </list>
+        /// </exception>
+        public virtual async Task DeadLetterMessageAsync(
+            ServiceBusReceivedMessage message,
+            IDictionary<string, object> propertiesToModify,
+            string deadLetterReason,
+            string deadLetterErrorDescription = default,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(message, nameof(message));
+            Argument.AssertNotNull(propertiesToModify, nameof(propertiesToModify));
+
+            // Prevent properties and arguments from setting distinct deadletter reasons or error descriptions
+            foreach (KeyValuePair<string, object> pair in propertiesToModify)
+            {
+                // Attempting to set the dead letter reason or description header through the properties
+                var isReasonHeaderKey = (pair.Key == AmqpMessageConstants.DeadLetterReasonHeader);
+                var isDescriptionHeaderKey = (pair.Key == AmqpMessageConstants.DeadLetterErrorDescriptionHeader);
+
+                if ((isReasonHeaderKey && deadLetterReason != null) && (pair.Value.ToString() != deadLetterReason))
+                {
+                    throw new ArgumentException("Differing deadletter reasons cannot be passed through both the properties and the parameter.");
```

Comment:
nit: I think InvalidOperationException is more intuitive here.

Reply:
Consider adding another sentence that says something like "pass it in one or the other place".